crates/nu-cli/src/repl.rs

```diff
@@ -161,7 +199,7 @@ pub fn evaluate_repl(
     if load_std_lib.is_none() && engine_state.get_config().show_banner {
         eval_source(
             engine_state,
-            stack,
+            stack.get_unique_mut(),
```

Comment:
the general pattern here: `stack.get_unique_mut()` ensures we are editing this stack in particular, and `stack.child_stack()` gives us a new stack with read-only access to the underlying stack.